src/Server/RestRouterHandlers/ColumnRestRouterHandler.cpp

```diff
@@ -154,7 +167,20 @@ std::pair<bool, String> ColumnRestRouterHandler::assertColumnExists(const String
 std::pair<bool, String> ColumnRestRouterHandler::assertColumnNotExists(const String & table, const String & column) const
 {
     const auto & catalog_service = CatalogService::instance(query_context);
-    auto [table_exist, column_exist] = catalog_service.columnExists(database, table, column);
+
+    bool table_exist, column_exist;
+    if (isDistributedDDL())
+    {
+        auto result = catalog_service.columnExists(database, table, column);
+        table_exist = std::get<0>(result);
+        column_exist = std::get<1>(result);
+    }
+    else
```

Comment:
Could we add "IF EXIST" in the second calling rather than checking it again by localColumnExists, which will make the logic simple.

Reply:
fixed